app/serializers/application.js

```diff
@@ -1,14 +1,36 @@
-import { ActiveModelSerializer } from 'active-model-adapter';
+import { ActiveModelSerializer } from "active-model-adapter";
 
 // Polymorphic associations are not supported in ember-data beta version:
 // refer: https://github.com/emberjs/data/issues/1574
 
-export default ActiveModelSerializer.extend({
+function normalize(payload) {
+  const key =
+    (payload.hasOwnProperty("messages") && "messages") ||
+    (payload.hasOwnProperty("message") && "message");
```

Comment:
_technically_ nothing prevents the API from sending the `message` AND the `messages` keys.

Maybe something like `_.flatten([ payload['message'], payload['messages'] ]).filter(_.identity).forEach(() => do something)